internal/core/src/storage/HuaweiCloudCredentialsProvider.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License
+
+#include <aws/core/auth/AWSCredentialsProvider.h>
+#include "HuaweiCloudSTSClient.h"
+
+namespace Aws {
+namespace Auth {
+class HuaweiCloudSTSAssumeRoleWebIdentityCredentialsProvider
```

Comment:
the implemention simlar with `AliyunSTSAssumeRoleWebIdentityCredentialsProvider`/`TencentCloudSTSAssumeRoleWebIdentityCredentialsProvider`

can we make those class as the template class ?